kong/plugins/grpc-gateway/deco.lua

```diff
@@ -199,6 +200,39 @@ local function unframe(body)
   return body:sub(pos, frame_end), body:sub(frame_end + 1)
 end
 
+--[[
+  // Set value `v` to table `t` at address stored in `key_stack`,
+  // Key stack is alternative to dot-syntax. For example:
+  // `a.b.c` is equivalent to `key_stack = { a, b, c }`.
+]]
+local function patch_table( t, key_stack, v )
+  local k = table.remove( key_stack, 1 ) -- pop out first key
+
+  if key_stack[1] == nil then --same as `if #key_stack == 0` but faster
+    t[k] = v
+  else 
+    if t[k] == nil then
+      t[k] = {}
+    end
+
+    patch_table( t[k], key_stack, v )
+  end
+end
+
+--[[
+  // Set value `v` at `path` in table `t`
+  // Path contains value address in dot-syntax. For example:
+  // `path="a.b.c"` would lead to `t[a][b][c] = v`.
+]]
+local function add_to_table( t, path, v )
+  local key_stack = {}
+
+  for captures in re_gmatch( path , "[^%.]+") do
+    table.insert( key_stack, captures[0] )
+  end
```

Comment:
> Hm... I was redirected here from https://github.com/Kong/kong-plugin-grpc-gateway. How many Kong repos are there? :)

Sorry for the noise, I was confused by some naming.  This is the right place.

I'm not too happy about constructing a list of segments just to immediately iterate on it, recursively and by removing/shifting from the start.  All this could be resumed with a single iteration, like this:  (also, the regex syntax was a mix of PCRE and Lua patterns)

```suggestion
local function add_to_table( t, path, v )
  local tab = t
  for m in re_gmatch( path , "([^.]+)(\\.)?") do
    local key, dot = m[1], m[2]

    if dot then
      tab[key] = tab[key] or {}
      tab = tab[key]
    else
      tab[key] = v
    end

  end
  return t
end
```

Reply:
I really like the way you simplified my code.

Reply:
my pleasure!  but it seems I didn't select the whole block and now there's a couple lines loose...

Reply:
should be fixed by now